include/mtxclient/http/client.hpp

```diff
@@ -761,6 +762,11 @@ class Client : public std::enable_shared_from_this<Client>
     //! Sets, updates, or deletes a pusher
     void set_pusher(const mtx::requests::SetPusher &req, Callback<mtx::responses::Empty> cb);
 
+    //! Searches the user directory
+    void search_user_directory(const std::string &search_term,
+                               int limit,
```

Comment:
this doesn't seem to be a required parameter. You could move it after the callback and give it a default value (or make it an optional).